test/src/androidTest/kotlin/io/realm/shared/notifications/RealmObjectNotificationsTests.kt

```diff
@@ -0,0 +1,179 @@
+            obj.update {
+                stringField = "Baz"
+            }
+            assertEquals("Baz", c.receive()!!.stringField)
+            observer.cancel()
+            c.close()
+        }
+    }
+
+    @Test
+    override fun cancelObserve() {
+        runBlocking {
+            val obj: Sample = realm.write {
+                copyToRealm(Sample().apply { stringField = "Foo" })
+            }
+            val c1 = Channel<Sample?>(1)
+            val c2 = Channel<Sample?>(1)
+            val observer1 = async {
+                obj.observe().collect {
+                    c1.trySend(it)
+                }
+            }
+            val observer2 = async {
+                obj.observe().collect {
+                    c2.trySend(it)
+                }
+            }
+            obj.update {
+                stringField = "Bar"
+            }
+            assertEquals("Bar", c1.receive()!!.stringField)
+            assertEquals("Bar", c2.receive()!!.stringField)
+            observer1.cancel()
+            obj.update {
+                stringField = "Baz"
+            }
+            assertEquals("Baz", c2.receive()!!.stringField)
+            assertTrue(c1.isEmpty)
+            observer2.cancel()
+            c1.close()
+            c2.close()
+        }
+    }
+
+    @Test
+    override fun initialElement() {
+        runBlocking {
+            val c = Channel<Sample?>(1)
+            val obj = realm.write {
+                copyToRealm(
+                    Sample().apply {
+                        stringField = "Foo"
+                    }
+                )
+            }
+            val observer = async {
+                obj.observe().collect {
+                    c.trySend(it)
+                }
+            }
+            assertEquals("Foo", c.receive()!!.stringField)
+            observer.cancel()
+            c.close()
+        }
+    }
+
+    @Test
+    override fun deleteObservable() {
+        runBlocking {
+            val c = Channel<Sample?>(1)
+            val obj: Sample = realm.write {
+                copyToRealm(Sample().apply { stringField = "Foo" })
+            }
+            val observer = async {
+                obj.observe().collect {
+                    c.trySend(it)
+                }
+                // Emit sentinel value to signal that flow completed
+                c.send(Sample())
+            }
+            assertNotNull(c.receive())
+            realm.write {
+                delete(findLatest(obj)!!)
+            }
+            assertNull(c.receive()) // Null is sent when object is deleted
+            assertEquals(Sample().stringField, c.receive()!!.stringField) // Test for sentinel value
+            observer.cancel()
+            c.close()
+        }
+    }
+
+    @Test
+    @Ignore
+    override fun closeRealmInsideFlowThrows() {
+        TODO("Not yet implemented")
+    }
+
+    @Test
+    override fun closingRealmDoesNotCancelFlows() {
```

Comment:
This leads to undefined behaviour in Core, since we still emit managed objects accessing a closed Realm!

Reply:
The test is mostly here so we capture whatever behavior we want. Right now it just stops emitting events (because change listeners stop), but the flows themselves are not completed.

We need to figure out exactly what should happen here. For writers we throw an exception, but I'm not sure that is possible for Flows since the even passed a few event queues along the way.